### ISSUE TITLE
nodeos: v1.8.7，Unable to use --snapshot synchronize data。v 1.8.6 Can synchronize normally

### ISSUE THREAD

**Issue by shiruitao** [EOSIO/eos#8371]
snap_2019-12-23-08-00.tar.gz
```
nodeos --config-dir /eos/eos/config --data-dir /eos/eos --snapshot /eos/eos/snapshots/snapshot-05bf86290fcbd129c5741d78673e080e2f4a564d2f59de1e36e0561150cf67ce.bin
```

All the time：
```
cleos get info
{
  "server_version": "f545d4e5",
  "chain_id": "aca376f206b8fc25a6ed44dbdc66547c36c6c33e3a119ffbeaef943642f0e906",
  "head_block_num": 96437801,
  "last_irreversible_block_num": 96437801,
  "last_irreversible_block_id": "05bf86290fcbd129c5741d78673e080e2f4a564d2f59de1e36e0561150cf67ce",
  "head_block_id": "05bf86290fcbd129c5741d78673e080e2f4a564d2f59de1e36e0561150cf67ce",
  "head_block_time": "2019-12-23T08:00:00.000",
  "head_block_producer": "atticlabeosb",
  "virtual_block_cpu_limit": 200000,
  "virtual_block_net_limit": 1048576000,
  "block_cpu_limit": 200000,
  "block_net_limit": 1048576,
  "server_version_string": "v1.8.7",
  "fork_db_head_block_num": 96437801,
  "fork_db_head_block_id": "05bf86290fcbd129c5741d78673e080e2f4a564d2f59de1e36e0561150cf67ce"
}
```
There are no errors in the logs。
```
info  2019-12-23T08:58:07.474 nodeos    main.cpp:93                   main                 ] nodeos version v1.8.7
info  2019-12-23T08:58:07.474 nodeos    main.cpp:94                   main                 ] nodeos using configuration file /eos/eos/config/config.ini
info  2019-12-23T08:58:07.474 nodeos    main.cpp:95                   main                 ] nodeos data directory is /eos/eos
info  2019-12-23T08:58:07.474 nodeos    controller.cpp:2261           startup              ] Starting initialization from snapshot, this may take a significant amount of time
info  2019-12-23T09:00:42.423 nodeos    controller.cpp:688            init                 ] database initialized with hash: 457e7eaeb5c69a2e3b6ee10141d4c715fd6cf1d88f1eb2887c028d15c62b466e
info  2019-12-23T09:00:42.423 nodeos    controller.cpp:2271           startup              ] Finished initialization from snapshot
info  2019-12-23T09:00:42.423 nodeos    chain_plugin.cpp:982          plugin_startup       ] starting chain in read/write mode
info  2019-12-23T09:00:42.423 nodeos    chain_plugin.cpp:986          plugin_startup       ] Blockchain started; head block is #96437801, genesis timestamp is 2018-06-08T08:08:08.888
info  2019-12-23T09:00:42.423 nodeos    http_plugin.cpp:538           plugin_startup       ] start listening for http requests
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/node/get_supported_apis
info  2019-12-23T09:00:42.424 nodeos    history_api_plugin.cpp:38     plugin_startup       ] starting history_api_plugin
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/history/get_actions
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/history/get_controlled_accounts
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/history/get_key_accounts
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/history/get_transaction
info  2019-12-23T09:00:42.424 nodeos    chain_api_plugin.cpp:77       plugin_startup       ] starting chain_api_plugin
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/abi_bin_to_json
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/abi_json_to_bin
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_abi
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_account
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_activated_protocol_features
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_block
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_block_header_state
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_code
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_code_hash
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_currency_balance
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_currency_stats
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_info
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_producer_schedule
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_producers
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_raw_abi
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_raw_code_and_abi
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_required_keys
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_scheduled_transactions
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_table_by_scope
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_table_rows
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/get_transaction_id
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/push_block
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/push_transaction
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/push_transactions
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/chain/send_transaction
info  2019-12-23T09:00:42.424 nodeos    net_api_plugin.cpp:63         plugin_startup       ] starting net_api_plugin
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/net/connect
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/net/connections
info  2019-12-23T09:00:42.424 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/net/disconnect
info  2019-12-23T09:00:42.425 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/net/status
info  2019-12-23T09:00:42.425 nodeos    http_plugin.cpp:625           add_handler          ] add api url: /v1/db_size/get
info  2019-12-23T09:00:42.425 nodeos    producer_plugin.cpp:833       plugin_startup       ] producer plugin:  plugin_startup() begin
info  2019-12-23T09:00:42.425 nodeos    producer_plugin.cpp:868       plugin_startup       ] producer plugin:  plugin_startup() end
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: api-full1.eoseoul.io port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: api-full2.eoseoul.io port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: boot.eostitan.com port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: bp.cryptolions.io port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: bp.eosbeijing.one port: 8080 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: bp.libertyblock.io port: 9800 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: br.eosrio.io port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: eos-seed-de.privex.io port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: fullnode.eoslaomao.com port: 443 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: mainnet.eoscalgary.io port: 5222 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: node.eosflare.io port: 1883 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: node1.eoscannon.io port: 59876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: p2p.eosdetroit.io port: 3018 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: p2p.genereos.io port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peer.eosn.io port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peer.main.alohaeos.com port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peer1.mainnet.helloeos.com.cn port: 80 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peer2.mainnet.helloeos.com.cn port: 80 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peering.mainnet.eoscanada.com port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peering1.mainnet.eosasia.one port: 80 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peering2.mainnet.eosasia.one port: 80 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: publicnode.cypherglass.com port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: seed1.greymass.com port: 9876 
info  2019-12-23T09:00:42.425 nodeos    net_plugin.cpp:1900           operator()           ] host: seed2.greymass.com port: 9876 
info  2019-12-23T09:01:12.424 nodeos    net_plugin.cpp:1900           operator()           ] host: api-full1.eoseoul.io port: 9876 
info  2019-12-23T09:01:12.424 nodeos    net_plugin.cpp:1900           operator()           ] host: api-full2.eoseoul.io port: 9876 
info  2019-12-23T09:01:12.424 nodeos    net_plugin.cpp:1900           operator()           ] host: boot.eostitan.com port: 9876 
info  2019-12-23T09:01:12.424 nodeos    net_plugin.cpp:1900           operator()           ] host: bp.cryptolions.io port: 9876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: bp.eosbeijing.one port: 8080 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: bp.libertyblock.io port: 9800 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: br.eosrio.io port: 9876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: eos-seed-de.privex.io port: 9876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: fullnode.eoslaomao.com port: 443 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: mainnet.eoscalgary.io port: 5222 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: node.eosflare.io port: 1883 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: node1.eoscannon.io port: 59876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: p2p.eosdetroit.io port: 3018 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: p2p.genereos.io port: 9876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peer.eosn.io port: 9876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peer1.mainnet.helloeos.com.cn port: 80 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peer2.mainnet.helloeos.com.cn port: 80 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peering1.mainnet.eosasia.one port: 80 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: peering2.mainnet.eosasia.one port: 80 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: publicnode.cypherglass.com port: 9876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: seed1.greymass.com port: 9876 
info  2019-12-23T09:01:12.425 nodeos    net_plugin.cpp:1900           operator()           ] host: seed2.greymass.com port: 9876
```

**Comment by matthewdarwin** [comment]
There are very few valid peers these days, check https://validate.eosnation.io/eos/reports/config.html

**Comment by wangke0809** [comment]
mee too! 12 hours have passed and no block has been received.  I tried again and it still didn't work.

```
  "server_version": "f545d4e5",
  "chain_id": "aca376f206b8fc25a6ed44dbdc66547c36c6c33e3a119ffbeaef943642f0e906",
  "head_block_num": 96755658,
  "last_irreversible_block_num": 96755658,
  "last_irreversible_block_id": "05c45fcad0cbc59584169665540d819031062cf109aebb590bd5c945a586f47a",
  "head_block_id": "05c45fcad0cbc59584169665540d819031062cf109aebb590bd5c945a586f47a",
  "head_block_time": "2019-12-25T05:00:00.000",
  "head_block_producer": "hashfineosio",
  "virtual_block_cpu_limit": 200000,
  "virtual_block_net_limit": 1048576000,
  "block_cpu_limit": 200000,
  "block_net_limit": 1048576,
  "server_version_string": "v1.8.7",
  "fork_db_head_block_num": 96755658,
  "fork_db_head_block_id": "05c45fcad0cbc59584169665540d819031062cf109aebb590bd5c945a586f47a"
}
```

**Comment by shiruitao** [comment]
> mee too! 12 hours have passed and no block has been received. I tried again and it still didn't work.
> 
> ```
>   "server_version": "f545d4e5",
>   "chain_id": "aca376f206b8fc25a6ed44dbdc66547c36c6c33e3a119ffbeaef943642f0e906",
>   "head_block_num": 96755658,
>   "last_irreversible_block_num": 96755658,
>   "last_irreversible_block_id": "05c45fcad0cbc59584169665540d819031062cf109aebb590bd5c945a586f47a",
>   "head_block_id": "05c45fcad0cbc59584169665540d819031062cf109aebb590bd5c945a586f47a",
>   "head_block_time": "2019-12-25T05:00:00.000",
>   "head_block_producer": "hashfineosio",
>   "virtual_block_cpu_limit": 200000,
>   "virtual_block_net_limit": 1048576000,
>   "block_cpu_limit": 200000,
>   "block_net_limit": 1048576,
>   "server_version_string": "v1.8.7",
>   "fork_db_head_block_num": 96755658,
>   "fork_db_head_block_id": "05c45fcad0cbc59584169665540d819031062cf109aebb590bd5c945a586f47a"
> }
> ```

use nodeos v1.8.6

**Comment by wangke0809** [comment]
@shiruitao 

TRX！我下午也用了 v1.8.6，发现可以，就是同步区块有点慢，一直落后 1000 个块左右。还在解决中...

**Comment by shiruitao** [comment]
@wangke0809 
p2p 节点的问题，我北京的服务器一直落后，香港的就没问题。

**Comment by ZhaoUncle** [comment]
新加坡节点用 1.8.7 也是不行，换成 1.8.6就可以了

**Comment by igorls** [comment]
I can confirm that booting from a snapshot without state and block.log does not work on v1.8.7
working fine on v2.0.0-rc3

**Comment by wangke0809** [comment]
@shiruitao 
请问一下节点服务器使用的什么配置啊？我 4 核 CPU 16 G RAM，开启了 history_plugin ，在美国地区和香港地区，节点都同步不到最新高度。

The node server has 4 CPUs and 16G RAM. I tried the servers in the United States and Hong Kong, and they could not sync to the latest block.

**Comment by shiruitao** [comment]
@wangke0809

**Comment by wangke0809** [comment]
@shiruitao 

解决没有呀，我增大机器磁盘 IO 之后可以同步到最新高度了，但是一段时间之后就跟不上了。给你的Gmail发了邮件，交流一下啊...

**Comment by gleehokie** [comment]
Fixed by #8380, thanks for the issue report @shiruitao and others.
